classes/ActionScheduler_Versions.php

```diff
@@ -115,6 +115,58 @@ public static function initialize_latest_version() {
 	 * @return string
 	 */
 	public function active_source() {
+		$file   = __FILE__;
+		$dir    = __DIR__;
+		$path   = $this->active_source_path();
+		$themes = (array) search_theme_directories();
+
+		foreach ( $themes as $slug => $data ) {
```

Comment:
I strongly suspect themes are generally the least common consumers of Action Scheduler. Could it make sense to evaluate this possibility *after* we've looped through the plugins?

Reply:
I decided to put themes first, as it seemed the least costly to check. Many sites only have a handful of themes, while plugins (especially WooCommerce sites) can have dozens. If you have a sample WooCommerce site you test with, would be interesting to see data indicating the performance of each section.

Reply:
> As it seemed the least costly to check.

Probably this is true ... but it also seems far less likely to yield a positive result. 

Suppose the cost of looping over all the themes is typically 5 units, and the cost of looping through all the plugins is typically 20 units. If in 99.% of cases Action Scheduler will A) never be found in the theme loop but B) will be found before we reach the end of the plugin loop, allowing us to bail out early, I think it makes sense to prioritize the plugin loop.

Reply:
My figures and units are obviously both imaginary. I can't recall a bug ticket or support case where we were working with a theme, though, whereas a great many have surfaced in relation to plugins.